### PR TITLE
glow: add new package

### PR DESCRIPTION
**Pull request by harrysharma1**
https://github.com/charmbracelet/glow/releases

### CODE REVIEW

**Comment by alecbcs** [comment]
@harrysharma1 can you include a link to the release/release notes in the PR description?

File: var/spack/repos/builtin/packages/glow/package.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    license("MIT")
+
+    version("1.5.1", sha256="d224be006fc4ee361c1d1a85dbd46c269372b76c5413e2a9bfd0f7fe52c66f78")
+
+    def install(self, spec, prefix):
+        mkdirp(prefix.bin)
+        install("glow", prefix.bin)
+        mkdirp(prefix.completions)
+        install_tree(".", prefix.completions)
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/glow/package.py]:
```suggestion
class Glow(GoPackage):
    """
    Glow is a terminal based markdown reader designed
    from the ground up to bring out the beauty—and power—of the CLI.
    Use it to discover markdown files,
    read documentation directly on the command line and stash markdown files
    to your own private collection, so you can read them anywhere.
    Glow will find local markdown files in subdirectories or a local Git repository.
    """

    homepage = "https://github.com/charmbracelet/glow"
    url = "https://github.com/charmbracelet/glow/archive/refs/tags/v1.5.1.tar.gz"

    license("MIT")

    version("1.5.1", sha256="b4ecf269b7c6447e19591b1d23f398ef2b38a6a75be68458390b42d3efc44b92")

    def install(self, spec, prefix):
        mkdirp(prefix.bin)
        install("glow", prefix.bin)
        mkdirp(prefix.completions)
        install_tree(".", prefix.completions)
```

Looks like right now this is just pulling the prebuilt linux-x86_64 glow executable. To make this package compatible with other OSs and architectures let's switch it over to being built from source as a GoPackage.

**Review by alecbcs — CHANGES_REQUESTED** [review]
@harrysharma1 thanks for the description update! One quick suggestion to move this package from downloading a prebuilt binary to building the binary from source for greater compatibility.

**Comment by harrysharma1** [comment]
Is there a way for spack to have multiple prebuilt binaries and for it to select based on a system. I know that it is probably more of a hassle to do than just building from the source, I was just curious.

**Comment by alecbcs** [comment]
There's not an easy way to pull binaries (and we strongly recommend against it unless that's your only option) but yes if you're interested in a package that adaptively pulls binaries for multiple architectures look at [`rust-bootstrap`](https://github.com/spack/spack/blob/097c04195a6d4f5102aada103c18a45394cfb7e9/var/spack/repos/builtin/packages/rust-bootstrap/package.py#L4).

**Review by alecbcs — APPROVED** [review]
Confirmed SHAs. Looks good to me. @harrysharma1 thanks for the quick turn around on the proposed changes.

**Pull request merged**